Connection/Connection.php

```diff
@@ -337,14 +337,33 @@ public function disconnect()
 
     /**
      * Set socket.
+     * A wrapper can be registered to handle application layer transport
      *
-     * @param   string  $socket    Socket URI.
+     * @param   string  $socketUri    Socket URI.
      * @return  \Hoa\Socket
+     * @throws  \Hoa\Socket\Exception
      */
-    protected function setSocket($socket)
-    {
+    protected function setSocket($socketUri)
+    {
+        $parsed = parse_url($socketUri);
+        if ($parsed !== false &&
+            isset($parsed['scheme']) &&
+            null !== $wrapper = Socket\Transport::getWrapper($parsed['scheme'])
+            ) {
+            $socket = $wrapper($socketUri);
```

Comment:
For `[$class, $method]`, yes it works, but for `'function'` I am not sure.
However, I don't think this particular case need to be addressed. And we filter with `callable` so it's fine.